### PR TITLE
not found page component

### PR DESCRIPTION
**Pull request by yousrakhaleel**
Relates#69

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -118,7 +126,7 @@ class App extends Component {
         })
         .catch((err) => {
           if (err.response.status === 500) {
-            window.location.href = '/error';
+            window.location.href = <serverError />;
```
Comment (zaher-aa) [inline client/src/App.jsx:129]: Why did you do that!?
It is not possible to assign a React component to a `window.location.href`
What we're doing here `window.location.href = '/error';` is converting the current window location to an endpoint called `/error` which is handled by React Router Dom for displaying a 500 - Server Error component

File: client/src/App.jsx [hunk 2]
```diff
@@ -141,7 +149,7 @@ class App extends Component {
...
```
Comment (zaher-aa) [inline client/src/App.jsx:152]: Same issue here

File: client/src/App.jsx [hunk 3]
```diff
@@ -174,7 +182,7 @@ class App extends Component {
       })
       .catch((err) => {
         if (err.response.status === 500) {
-          window.location.href = '/error';
+          window.location.href = <serverError />;
```
Comment (zaher-aa) [inline client/src/App.jsx:185]: Here also...

File: client/src/App.jsx [hunk 4]
```diff
@@ -264,8 +272,8 @@ class App extends Component {
               path="product/:id"
               element={<ProductDetails addToCart={this.addToCart} />}
             />
-            <Route path="/error" element={'Server Error'} />
-            <Route path="*" element={'Page Not Found'} />
+            <Route path="/error" element={<serverError />} />
```
Comment (zaher-aa) [inline client/src/App.jsx:275]: Good job here.
Just edit the `serverError` and `notFound` to use PascalCase ==> `ServerError', `NotFound`

File: client/src/components/NotFound/index.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import notFound from "./notFound";
```
Comment (zaher-aa) [inline client/src/components/NotFound/index.js:1]: Also edit the files name to use PascalCase

**Review by zaher-aa — CHANGES_REQUESTED** [review]

**Review by zaher-aa — APPROVED** [review]

**Review by Fady-Alwazir — APPROVED** [review]

**Pull request merged**